### PR TITLE
 proper ssl implementation, handshake is done by haxe.ssl.Socket itself

### PR DESCRIPTION
**Pull request by tecteun**

### CODE REVIEW

File: src/haxe/net/impl/WebSocketGeneric.hx [hunk 1]
```diff
@@ -380,7 +380,9 @@ class WebSocketGeneric extends WebSocket {
 
     private function prepareFrame(data:Bytes, type:Opcode, isFinal:Bool):Bytes {
         var out = new BytesRW();
-        var isMasked = true; // All clientes messages must be masked: http://tools.ietf.org/html/rfc6455#section-5.1
+	
+        //Chrome: VM321:1 WebSocket connection to 'ws://localhost:8000/' failed: A server must not mask any frames that it sends to the client.
+        var isMasked = false; //true; // All clientes messages must be masked: http://tools.ietf.org/html/rfc6455#section-5.1
```
Comment (soywiz) [inline src/haxe/net/impl/WebSocketGeneric.hx]: So this is generating a client frame? Then shouldn't it be masked?

**Comment by soywiz** [comment]
Is this ready?

**Comment by tecteun** [comment]
yep, this is ready to merge.
I've found that the ssl handshake etc is done in sys.net.sslsocket itself. So I've removed that. 
I'm not really sure about the "masking" thing, though without this fix chrome is complaining that messages should not be masked, so I've left that as is.

**Comment by soywiz** [comment]
Nice. Thanks!

**Pull request merged**